main/src/com/google/refine/model/recon/StandardReconConfig.java

```diff
@@ -475,51 +463,40 @@ private CloseableHttpClient getHttpClient() {
-                new BasicNameValuePair("queries", queriesString));
-        request.setEntity(new UrlEncodedFormEntity(body, Consts.UTF_8));
-        
-        try (CloseableHttpResponse response = getHttpClient().execute(request)) {
-            StatusLine statusLine = response.getStatusLine();
-            if (statusLine.getStatusCode() >= 400) {
-                logger.error("Failed  - code: "
-                        + Integer.toString(statusLine.getStatusCode())
-                        + " message: " + statusLine.getReasonPhrase());
+        try {
+            String responseString = postQueries(service, queriesString);
+            ObjectNode o = ParsingUtilities.evaluateJsonStringToObjectNode(responseString);
+
+            if (o == null) { // utility method returns null instead of throwing
+                logger.error("Failed to parse string as JSON: " + responseString);
             } else {
-                String s = ParsingUtilities.inputStreamToString(response.getEntity().getContent());
-                ObjectNode o = ParsingUtilities.evaluateJsonStringToObjectNode(s);
-                if (o == null) { // utility method returns null instead of throwing
-                    logger.error("Failed to parse string as JSON: " + s);
-                } else {
-                    for (int i = 0; i < jobs.size(); i++) {
-                        StandardReconJob job = (StandardReconJob) jobs.get(i);
-                        Recon recon = null;
-
-                        String text = job.text;
-                        String key = "q" + i;
-                        if (o.has(key) && o.get(key) instanceof ObjectNode) {
-                            ObjectNode o2 = (ObjectNode) o.get(key);
-                            if (o2.has("result") && o2.get("result") instanceof ArrayNode) {
-                                ArrayNode results = (ArrayNode) o2.get("result");
-
-                                recon = createReconServiceResults(text, results, historyEntryID);
-                            } else {
-                                logger.warn("Service error for text: " + text + "\n  Job code: " + job.code + "\n  Response: " + o2.toString());
-                            }
+                for (int i = 0; i < jobs.size(); i++) {
+                    StandardReconJob job = (StandardReconJob) jobs.get(i);
+                    Recon recon = null;
+
+                    String text = job.text;
+                    String key = "q" + i;
+                    if (o.has(key) && o.get(key) instanceof ObjectNode) {
+                        ObjectNode o2 = (ObjectNode) o.get(key);
+                        if (o2.has("result") && o2.get("result") instanceof ArrayNode) {
+                            ArrayNode results = (ArrayNode) o2.get("result");
+
+                            recon = createReconServiceResults(text, results, historyEntryID);
                         } else {
-                            // TODO: better error reporting
```

Comment:
I'd also leave this TODO since error reporting should not be in the server logs: users should be notified about it.

Reply:
I think you are just seeing a diff artifact. The exact same text is a couple of lines down, but I've added a second copy.